low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -116,440 +305,127 @@ def initialize(self, conf_reader, msgQlist, product):
         self.iem.check_exsisting_fault_iems()
         self.KAFKA = self.iem.EVENT_CODE["KAFKA_ACTIVE"][1]
 
-        # Integrate into the main dbus loop to catch events
-        DBusGMainLoop(set_as_default=True)
-
-        # Initialize SystemBus and get Manager Interface
-        self._bus = SystemBus()
-        systemd = self._bus.get_object("org.freedesktop.systemd1",
-                                       "/org/freedesktop/systemd1")
-        self._manager = Interface(systemd,
-                            dbus_interface='org.freedesktop.systemd1.Manager')
-
-        self._node_id = Conf.get(GLOBAL_CONF, NODE_ID_KEY,'SN01')
-
-        cache_dir_path = os.path.join(DATA_PATH, self.CACHE_DIR_NAME)
-        self.SERVICE_MONITOR_DATA_PATH = os.path.join(cache_dir_path,
-                                         f'SERVICE_MONITOR_DATA_{self._node_id}')
-        # Get the stored previous service info
-        self.persistent_service_data = {}
-        if os.path.isfile(self.SERVICE_MONITOR_DATA_PATH):
-            self.persistent_service_data = \
-                store.get(self.SERVICE_MONITOR_DATA_PATH)
-        if self.persistent_service_data:
-            self.not_active_services = \
-                self.persistent_service_data['not_active_services']
-            self.failed_services = \
-                self.persistent_service_data['failed_services']
-            self.service_status = \
-                self.persistent_service_data['service_status']
-        else:
-            self.persistent_service_data = {
-                'not_active_services' : self.not_active_services,
-                'failed_services' : self.failed_services,
-                'service_status' : self.service_status
-            }
-            store.put(self.persistent_service_data,
-                      self.SERVICE_MONITOR_DATA_PATH)
-
-        self.remove_disabled_services()
+        self.initialize_dbus()
+        for service in self.services_to_monitor:
+            self.initialize_service(service)
+        self.subscribe_unit_file_changed_signal()
```

Comment:
if initialize_dbus() had failed, this subscriptn will fail too, so need checks around

Reply:
This will fail sensor initialize & shutdown will be called. understood after our discussion